File: tests/uasm/sample/array_reassign_then_read/ArrayReassignThenReadTest.cs
using UdonSharp;
using UnityEngine;

[UdonBehaviourSyncMode(BehaviourSyncMode.None)]
public class ArrayReassignThenReadTest : UdonSharpBehaviour
{
    public void Start()
    {
        int[] values = new int[] { 2, 4, 6 };
        values[0] = values[1] + 1;
        values[2] = values[0] + values[1];

        Debug.Log(values[0]);
        Debug.Log(values[2]);
    }
}

File: tests/uasm/sample/array_reassign_then_read/ArrayReassignThenReadTest.ts
import { UdonBehaviour } from "@ootr/udon-assembly-ts/stubs/UdonDecorators";
import { UdonSharpBehaviour } from "@ootr/udon-assembly-ts/stubs/UdonSharpBehaviour";
import { Debug } from "@ootr/udon-assembly-ts/stubs/UnityTypes";

@UdonBehaviour()
export class ArrayReassignThenReadTest extends UdonSharpBehaviour {
  Start(): void {
    const values = [2, 4, 6];
    values[0] = values[1] + 1;
    values[2] = values[0] + values[1];

    Debug.Log(values[0]);
    Debug.Log(values[2]);
  }
}
